thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/formatter/channels/JiraContentFormatter.java

```diff
@@ -0,0 +1,150 @@
+
+  private static final Map<String, String> alertContentToTemplateMap;
+  static {
+    Map<String, String> aMap = new HashMap<>();
+    aMap.put(MetricAnomaliesContent.class.getSimpleName(), "jira-metric-anomalies-template.ftl");
+    alertContentToTemplateMap = Collections.unmodifiableMap(aMap);
+  }
+
+  public JiraContentFormatter(JiraConfiguration jiraAdminConfig, Properties jiraClientConfig,
+      BaseNotificationContent content, ThirdEyeAnomalyConfiguration teConfig, ADContentFormatterContext adContext) {
+    super(jiraClientConfig, content, teConfig, adContext);
+
+    this.jiraAdminConfig = jiraAdminConfig;
+    validateJiraConfigs(jiraAdminConfig);
+  }
+
+  /**
+   * Make sure the base admin parameters are configured before proceeding
+   */
+  private void validateJiraConfigs(JiraConfiguration jiraAdminConfig) {
+    Preconditions.checkNotNull(jiraAdminConfig.getJiraUser());
+    Preconditions.checkNotNull(jiraAdminConfig.getJiraPassword());
+    Preconditions.checkNotNull(jiraAdminConfig.getJiraHost());
+  }
+
+  public IssueInput getJiraEntity(Collection<AnomalyResult> anomalies) {
+    Map<String, Object> templateData = notificationContent.format(anomalies, adContext);
+    templateData.put("dashboardHost", teConfig.getDashboardHost());
+    return buildJiraEntity(alertContentToTemplateMap.get(notificationContent.getTemplate()), templateData);
+  }
+
+  /**
+   * Apply the parameter map to given email template, and format it as EmailEntity
+   */
+  private IssueInput buildJiraEntity(String jiraTemplate, Map<String, Object> templateValues) {
+    String issueSummary = BaseNotificationContent.makeSubject(getSubjectType(alertClientConfig), this.adContext.getNotificationConfig(), templateValues);
+
+    // Fetch the jira project and issue type fields if overridden by user
+    String jiraProject = MapUtils.getString(alertClientConfig, PROP_PROJECT, this.jiraAdminConfig.getJiraDefaultProjectKey());
+    Long jiraIssueTypeId = MapUtils.getLong(alertClientConfig, PROP_ISSUE_TYPE, this.jiraAdminConfig.getJiraIssueTypeId());
+
+    IssueInputBuilder issueBuilder = new IssueInputBuilder(jiraProject, jiraIssueTypeId, issueSummary);
+
+    String assignee = MapUtils.getString(alertClientConfig, PROP_ASSIGNEE);
+    if (StringUtils.isNotBlank(assignee)) {
+      LOG.info("Assigning the jira to " + alertClientConfig.get(PROP_ASSIGNEE));
+      issueBuilder.setAssigneeName(assignee);
+    }
+
+    List<String> labels = ConfigUtils.getList(alertClientConfig.get(PROP_LABELS));
+    labels.add(PROP_DEFAULT_LABEL);
+    issueBuilder.setFieldValue("labels", labels);
+
+    HtmlEmail email = new HtmlEmail();
+    String cid = "";
+    try {
+      if (StringUtils.isNotBlank(this.notificationContent.getSnaphotPath())) {
+        cid = email.embed(new File(this.notificationContent.getSnaphotPath()));
+      }
+    } catch (Exception e) {
+      LOG.error("Exception while embedding screenshot for anomaly", e);
```

Comment:
Looks like this only plugs in the snapshot. When is the screenshot gets generated?

Reply:
The screenshots are generated in the respective templates when you trigger notificationContent.format(...)